docs/local_development.md

```diff
@@ -0,0 +1,27 @@
+# Recommended Local Development and Debugging Workflow
+
+## Overview
+
+We recommend 2 modes of running the broker for development work.
+
+1. Build a local image and deploy from image
+2. Build the broker executable and run it locally connected to a local oc cluster up from [fusor/catasb 'dev' branch](https://github.com/fusor/catasb/tree/dev)
+
+## Build a local image and deploy from image
+
+1. Build the Ansible Service Broker executable with the command: ```make build```
+2. Edit the Makefile to reflect the BROKER_IMAGE_NAME and TAG you want to use for local development
+3. Build a development image:  ```make build-image```
+4. Deploy from [templates/deploy-ansible-service-broker.template.yaml](https://github.com/openshift/ansible-service-broker/blob/master/templates/deploy-ansible-service-broker.template.yaml) setting the parameter "-p BROKER_IMAGE" to your local image name/tag 
+
+## Running the broker executable locally
+
+1. Build the Ansible Service Broker executable with the command: ```make build```
+2. Ensure you have a local oc cluster up environment running with [fusor/catasb 'dev' branch](https://github.com/fusor/catasb/tree/dev)
+3. ```cd scripts```
+4. ```cp my_local_dev_vars.example my_local_dev_vars```
+5. Edit 'my_local_dev_vars' 
+6. Run: ```./run_local.sh```
+  * This will remove the running 'asb' pod and replace the endpoint of the asb route with the locally running broker executable. 
+7. Use the cluster as you normally would for testing.
+  * When you want to make a change and rebuild the broker.  CTRL-C ```run_local.sh```, rebuild and re-run ```run_local.sh```
```

Comment:
I really hate the need for the scripts. I would prefer `make run` handle all of this.